quickwit-backward-compat/tests/test_backward_compatibility.rs

```diff
@@ -36,6 +55,11 @@ fn test_index_metadata_deser(path: &Path) -> anyhow::Result<()> {
     let index_metadata = read_index_metadata(path)?;
     let expected_path = path.to_string_lossy().replace(".json", ".expected.json");
     let expected_index_metadata = read_index_metadata(Path::new(&expected_path))?;
+    test_index_metadata_eq(&index_metadata, &expected_index_metadata);
+    Ok(())
+}
+
+fn test_index_metadata_eq(index_metadata: &IndexMetadata, expected_index_metadata: &IndexMetadata) {
```

Comment:
This is brittle and will break when a field is added to `IndexMetadata`.

Reply:
We decided not to derive `PartialEq` on`IndexMetadata` because `SourceConfig` carries a `serde_json::Value`.

Reply:
serde_json::Value implements PartialEq and the default map implementation is a BTreeMap.
Was there a problem with using this directly?

Reply:
Discussed off GitHub: a few months ago, we decided that even if `serde_json::Value` implements `PartialEq`, that was a bit sketched. Today we are changing our minds.

Reply:
No go. `DocMapping` does not implement `PartialEq`.